src/jit/lowerarmarch.cpp

```diff
@@ -348,6 +343,66 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
     }
 }
 
+//------------------------------------------------------------------------
+// ContainBlockStoreAddress: Attempt to contain an address used by an unrolled block store.
+//
+// Arguments:
+//    blkNode - the block store node
+//    size - the block size
+//    addr - the address node to try to contain
+//
+void Lowering::ContainBlockStoreAddress(GenTreeBlk* blkNode, unsigned size, GenTree* addr)
+{
+    assert(blkNode->OperIs(GT_STORE_BLK) && (blkNode->gtBlkOpKind == GenTreeBlk::BlkOpKindUnroll));
+    assert(size < INT32_MAX);
+
+    if (addr->OperIsLocalAddr())
+    {
+        addr->SetContained();
+        return;
+    }
+
+    if (!addr->OperIs(GT_ADD) || addr->gtOverflow() || !addr->AsOp()->gtGetOp2()->OperIs(GT_CNS_INT))
+    {
+        return;
+    }
+
+    GenTreeIntCon* offsetNode = addr->AsOp()->gtGetOp2()->AsIntCon();
+    ssize_t        offset     = offsetNode->IconValue();
+
+    // All integer load/store instructions on both ARM32 and ARM64 support
+    // offsets in range -255..255. Of course, this is a rather conservative
+    // check. For example, if the offset and size are a multiple of 8 we
+    // could allow a combined offset of up to 32760 on ARM64.
+    if ((offset < -255) || (offset > 255) || (offset + static_cast<int>(size) > 256))
+    {
+        return;
+    }
+
+#ifdef _TARGET_ARM64_
+    // If we're going to use LDP/STP we need to ensure that the offset is
+    // a multiple of 8 since these instructions do not have an unscaled
+    // offset variant.
+    if ((size >= 2 * REGSIZE_BYTES) && (offset % REGSIZE_BYTES != 0))
+    {
+        return;
+    }
+#endif
+
+    if (!IsSafeToContainMem(blkNode, addr))
```

Comment:
From the description of `IsSafeToContainMem` it is not obvious that it allows non-immediate children, but from the code it is obvious that it does so calls like `ContainBlockStoreAddress(blkNode, size, src->AsIndir()->Addr());` are correct.

Reply:
Yeah, the parent/child naming doesn't make a lot of sense for `IsSafeToContainMem`. This more or less checks if it's safe to move a "childNode" forward to the "parentNode" point. The parent/child appeared because it was used for containment.